lib/opensusebasetest.pm

```diff
@@ -1049,13 +1049,22 @@ sub wait_boot_past_bootloader {
     my $nologin = $args{nologin};
     my $forcenologin = $args{forcenologin};
 
-    # Workaround for bsc#1204221 and bsc#1204230
+    # Workaround for bsc#1204221, bsc#1204230 and bsc#1203641
     if (is_sle('=15-SP5') && check_var('VIRSH_VMM_FAMILY', 'hyperv')) {
         # This should only happen on SLE15SP5 and on hyperv
-        record_soft_failure 'workaround bsc#1204221 - Failed to boot at SLES15SP5 with gnome in hyperv 2019' if check_var('DESKTOP', 'gnome');
-        record_soft_failure 'workaround bsc#1204230 - Failed to boot at SLES15SP5 with x server in hyperv 2019' if check_var('DESKTOP', 'textmode');
-        sleep 30;
-        send_key 'esc';
+        # This is for legacy issue.
+        if (check_var('UEFI', '0')) {
+            record_soft_failure 'workaround bsc#1204221 - Failed to boot at SLES15SP5 with gnome in hyperv 2019' if check_var('DESKTOP', 'gnome');
+            record_soft_failure 'workaround bsc#1204230 - Failed to boot at SLES15SP5 with x server in hyperv 2019' if check_var('DESKTOP', 'textmode');
+            sleep 30;
```

Comment:
Using "sleep" is not optimal. You should avoid such usage unless it is absolutely necessary or you do not have other measures to achieve what you want to do. Please refer to [code contribution](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md)